src/widget/scroll_view.rs

```diff
@@ -95,28 +100,73 @@ impl Widget for ScrollView {
         self.child.update(cx);
     }
 
-    fn measure(&mut self, cx: &mut LayoutCx) -> (Size, Size) {
-        let _ = self.child.measure(cx);
-        (Size::ZERO, Size::new(1e9, 1e9))
+    fn compute_max_intrinsic(&mut self, axis: Axis, cx: &mut LayoutCx, bc: &BoxConstraints) -> f64 {
+        match axis {
+            Axis::Horizontal => {
+                if bc.min().width.is_sign_negative() {
+                    0.0
+                } else {
+                    let length =
+                        self.child
+                            .compute_max_intrinsic(axis, cx, &bc.unbound_max_height());
+                    length.min(bc.max().width).max(bc.min().width)
+                }
+            }
+            Axis::Vertical => {
+                if bc.min().height.is_sign_negative() {
+                    0.0
+                } else {
+                    let length =
+                        self.child
+                            .compute_max_intrinsic(axis, cx, &bc.unbound_max_height());
+                    length.min(bc.max().height).max(bc.min().height)
+                }
+            }
+        }
     }
 
-    fn layout(&mut self, cx: &mut LayoutCx, proposed_size: Size) -> Size {
-        let child_proposed = Size::new(proposed_size.width, 1e9);
-        let child_size = self.child.layout(cx, child_proposed);
-        Size::new(child_size.width, proposed_size.height)
+    fn layout(&mut self, cx: &mut LayoutCx, bc: &BoxConstraints) -> Size {
+        cx.request_paint();
```

Comment:
Is this necessary even, when the layout hasn't changed at all?

(I see this is done in other widgets as well though, maybe a point to keep in mind later for optimization etc.)

Reply:
I'm not sure if this is the best way of doing it. But I was seeing the visual rendering on screen failing to update properly before I put this in.